### PR TITLE
#Issue #565 - initial creation of intersection type

### PR DESCRIPTION
**Pull request by kamilkrzywanski**
## What it does
fixes issue #565 

## Author checklist

- [x] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Comment by kamilkrzywanski** [comment]
@jarthana @mpalat Can you look at it?

**Comment by mpalat** [comment]
> @jarthana @mpalat Can you look at it?

@jarthana - since it's APT will prefer you to have a look

**Comment by kamilkrzywanski** [comment]
@jarthana can you review this change?

File: org.eclipse.jdt.apt.pluggable.tests/src/org/eclipse/jdt/apt/pluggable/tests/processors/buildertester/Issue565Processor.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+@SupportedSourceVersion(SourceVersion.RELEASE_6)
+public class Issue565Processor extends AbstractProcessor {
+	private static boolean status = false;
+
+	@Override
+	public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+
+		String classpath = System.getProperty("java.class.path");
+		String[] classPathValues = classpath.split(File.pathSeparator);
+		for (String classPath: classPathValues) {
```
Comment (jarthana) [inline org.eclipse.jdt.apt.pluggable.tests/src/org/eclipse/jdt/apt/pluggable/tests/processors/buildertester/Issue565Processor.java:41]: Is this really required?

File: org.eclipse.jdt.apt.pluggable.tests/src/org/eclipse/jdt/apt/pluggable/tests/processors/buildertester/Issue565Processor.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+			System.out.println(classPath);
+		}
+
+		for (Element element : roundEnv.getElementsAnnotatedWith(annotations.stream().findAny().get())) {
+				if (element instanceof TypeElement) {
```
Comment (jarthana) [inline org.eclipse.jdt.apt.pluggable.tests/src/org/eclipse/jdt/apt/pluggable/tests/processors/buildertester/Issue565Processor.java:46]: I am wondering we make the testing parameters a bit more elaborate? Looks like we aren't doing much with the visiting elements and all we check for is that getElementsAnnotatedWith() doesn't crash. We don't seem to be using what the keyBuilder is doing. Also, this for loop is not indented properly.

File: org.eclipse.jdt.apt.pluggable.tests/src/org/eclipse/jdt/apt/pluggable/tests/processors/buildertester/Issue565Processor.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+@SupportedAnnotationTypes("targets.issue565.Annotation565")
+@SupportedSourceVersion(SourceVersion.RELEASE_6)
+public class Issue565Processor extends AbstractProcessor {
+	private static boolean status = false;
```
Comment (jarthana) [inline org.eclipse.jdt.apt.pluggable.tests/src/org/eclipse/jdt/apt/pluggable/tests/processors/buildertester/Issue565Processor.java:34]: I see the process() being executed twice. Perhaps you can check processingOver() like in other processors to filter out the last round.

File: org.eclipse.jdt.core.compiler.batch/pom.xml [hunk 4]
```diff
@@ -17,7 +17,7 @@
     <version>4.33.0-SNAPSHOT</version>
   </parent>
   <artifactId>org.eclipse.jdt.core.compiler.batch</artifactId>
-  <version>3.39.0-SNAPSHOT</version>
+  <version>3.39.100-SNAPSHOT</version>
```
Comment (jarthana) [inline org.eclipse.jdt.core.compiler.batch/pom.xml:20]: This is not required as the minor version has already changed.

File: org.eclipse.jdt.core.compiler.batch/META-INF/MANIFEST.MF [hunk 5]
```diff
@@ -3,7 +3,7 @@ Main-Class: org.eclipse.jdt.internal.compiler.batch.Main
 Bundle-ManifestVersion: 2
 Bundle-Name: Eclipse Compiler for Java(TM)
 Bundle-SymbolicName: org.eclipse.jdt.core.compiler.batch
-Bundle-Version: 3.39.0.qualifier
+Bundle-Version: 3.39.100.qualifier
```
Comment (jarthana) [inline org.eclipse.jdt.core.compiler.batch/META-INF/MANIFEST.MF:6]: This is not required as the minor version has already changed.

File: org.eclipse.jdt.core/META-INF/MANIFEST.MF [hunk 6]
```diff
@@ -2,7 +2,7 @@ Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-Name: %pluginName
 Bundle-SymbolicName: org.eclipse.jdt.core; singleton:=true
-Bundle-Version: 3.39.0.qualifier
+Bundle-Version: 3.39.100.qualifier
```
Comment (jarthana) [inline org.eclipse.jdt.core/META-INF/MANIFEST.MF:5]: This is not required as the minor version has already changed.

File: org.eclipse.jdt.core/pom.xml [hunk 7]
```diff
@@ -17,7 +17,7 @@
     <version>4.33.0-SNAPSHOT</version>
   </parent>
   <artifactId>org.eclipse.jdt.core</artifactId>
-  <version>3.39.0-SNAPSHOT</version>
+  <version>3.39.100-SNAPSHOT</version>
```
Comment (jarthana) [inline org.eclipse.jdt.core/pom.xml:20]: This is not required as the minor version has already changed.

File: org.eclipse.jdt.apt.pluggable.core/META-INF/MANIFEST.MF [hunk 8]
```diff
@@ -2,13 +2,13 @@ Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-Name: %pluginName
 Bundle-SymbolicName: org.eclipse.jdt.apt.pluggable.core;singleton:=true
-Bundle-Version: 1.4.500.qualifier
+Bundle-Version: 1.4.600.qualifier
```
Comment (jarthana) [inline org.eclipse.jdt.apt.pluggable.core/META-INF/MANIFEST.MF:5]: This is not required as the minor version has already changed.

File: org.eclipse.jdt.apt.pluggable.tests/META-INF/MANIFEST.MF [hunk 9]
```diff
@@ -13,7 +13,7 @@ Require-Bundle: org.junit,
  org.eclipse.jdt.core.tests.builder,
  org.eclipse.jdt.core.tests.compiler,
  org.eclipse.test.performance,
- org.eclipse.jdt.core;bundle-version="3.38.0",
+ org.eclipse.jdt.core;bundle-version="[3.39.0,4.0.0)",
```
Comment (jarthana) [inline org.eclipse.jdt.apt.pluggable.tests/META-INF/MANIFEST.MF:16]: This is not required as the minor version has already changed.

**Review by jarthana — CHANGES_REQUESTED** [review]
I find the code changes themselves acceptable. But the bundle versions are not required and tests need some changes.

File: org.eclipse.jdt.apt.pluggable.tests/src/org/eclipse/jdt/apt/pluggable/tests/processors/buildertester/Issue565Processor.java (hunk 1 above)
Comment (kamilkrzywanski) [reply org.eclipse.jdt.apt.pluggable.tests/src/org/eclipse/jdt/apt/pluggable/tests/processors/buildertester/Issue565Processor.java]: My fault :) Removed

File: org.eclipse.jdt.apt.pluggable.tests/src/org/eclipse/jdt/apt/pluggable/tests/processors/buildertester/Issue565Processor.java [hunk 10]
```diff
@@ -35,14 +34,12 @@ public class Issue565Processor extends AbstractProcessor {
...
+        }
 
-		String classpath = System.getProperty("java.class.path");
-		String[] classPathValues = classpath.split(File.pathSeparator);
-		for (String classPath: classPathValues) {
-			System.out.println(classPath);
-		}
-
-		for (Element element : roundEnv.getElementsAnnotatedWith(annotations.stream().findAny().get())) {
+		for (Element element : roundEnv.getElementsAnnotatedWith(Annotation565.class)) {
```
Comment (jarthana) [inline org.eclipse.jdt.apt.pluggable.tests/src/org/eclipse/jdt/apt/pluggable/tests/processors/buildertester/Issue565Processor.java:42]: This is causing a compilation problem. The previous change was better, imo. Also, please correct the indentation, squash all the changes if possible. And please mark all the resolved comments as such.

**Review by jarthana — APPROVED** [review]

**Pull request merged**